deps/gcc.Dockerfile

```diff
@@ -1,151 +1,39 @@
-RUN wget http://ftp.gnu.org/gnu/gcc/${GCC}/${GCC}.tar.bz2  -O - | tar -xj
-RUN wget http://ftp.gnu.org/gnu/mpfr/${MPFR}.tar.xz  -O - | tar -xJ
-RUN wget http://ftp.gnu.org/gnu/mpc/${MPC}.tar.gz  -O - | tar -xz
-RUN wget http://ftp.gnu.org/gnu/gmp/${GMP}.tar.xz  -O - | tar -xJ
-RUN wget ftp://sourceware.org/pub/newlib/${NEWLIB}.tar.gz  -O - | tar -xz
-
-RUN apk add --no-cache bash build-base zlib-dev
-
-#################################################################################
-##########################    Building binutils    ##############################
-#################################################################################
-WORKDIR $BUILDDIR/$BINUTILS
-RUN $SRCDIR/$BINUTILS/configure \
-	--prefix=/usr \
-	--target=$TARGET \
-	--disable-multilib \
-	--with-cpu=$TARGET_CPU \
-	--disable-nls
+ARG CROSSTOOL_VERSION
 
-RUN make -j"$(nproc)"
-RUN make prefix=$INSTALLDIR/$BINUTILS/usr install
-RUN rm -rf $INSTALLDIR/$BINUTILS/usr/share
-RUN rm $INSTALLDIR/$BINUTILS/usr/lib/bfd-plugins/libdep.so
-RUN cp -r $INSTALLDIR/$BINUTILS/usr/ $STAGINGDIR
-RUN cp -r $INSTALLDIR/$BINUTILS/usr/* /usr/
 #################################################################################
-#########################    Building GCC bootstrap    ##########################
+##########################    Install required packages    ######################
 #################################################################################
-WORKDIR $SRCDIR/$GCC
-RUN ln -fs ../$MPFR mpfr
-RUN ln -fs ../$MPC mpc
-RUN ln -fs ../$GMP gmp
-
-WORKDIR $BUILDDIR/$GCC-bootstrap
-# These flags are required to build older versions of GCC
-ENV CFLAGS="$CFLAGS -Wno-implicit-fallthrough -Wno-cast-function-type -fpermissive"
-ENV CXXFLAGS="$CXXFLAGS -Wno-implicit-fallthrough -Wno-cast-function-type -fpermissive"
-RUN $SRCDIR/$GCC/configure \
-	--prefix=/usr \
-	--target=$TARGET \
-	--enable-languages="c" \
-	--disable-multilib \
-	--with-cpu=$TARGET_CPU \
-	--with-system-zlib \
-	--with-libgloss \
-	--without-headers \
-	--disable-shared \
-	--disable-nls
+RUN apk update && apk upgrade && apk add cmake alpine-sdk wget xz git bash autoconf automake bison flex texinfo help2man gawk libtool ncurses-dev gettext-dev python3-dev rsync openjdk11 ninja
 
-RUN make -j"$(nproc)" all-gcc
-RUN make DESTDIR="$INSTALLDIR/$GCC-bootstrap" install-strip-gcc
-RUN rm -rf "$INSTALLDIR/$GCC-bootstrap/usr/share"
-RUN cp -r "$INSTALLDIR/$GCC-bootstrap/usr/"* /usr/
 #################################################################################
-#############################    Building NewLib    #############################
+############################    Building crosstool-ng    ########################
 #################################################################################
-WORKDIR $BUILDDIR/$NEWLIB
-ENV CFLAGS_FOR_TARGET="-Os -g -ffunction-sections -fdata-sections -fomit-frame-pointer -ffast-math"
+WORKDIR /root
```

Comment:
Build crosstool-ng. Not currently supported as a package in this version of Alpine.